File: backend/src/main/java/com/pharmacy/medlan/enums/DosageForm.java
package com.pharmacy.medlan.enums;

public enum DosageForm {
    TABLET("Tablet"),
    CAPSULE("Capsule"),
    SYRUP("Syrup"),
    INJECTION("Injection"),
    DROPS("Drops"),
    CREAM("Cream"),
    OINTMENT("Ointment"),
    INHALER("Inhaler"),
    POWDER("Powder"),
    SUSPENSION("Suspension"),
    GEL("Gel"),
    LOTION("Lotion"),
    SPRAY("Spray"),
    PATCH("Patch"),
    SUPPOSITORY("Suppository");

    private final String displayName;

    DosageForm(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() { return displayName; }
}

File: frontend/src/constants/dosageForm.js
/**
 * Dosage Form - Matching backend DosageForm.java enum
 */
export const DOSAGE_FORM = {
  TABLET: "TABLET",
  CAPSULE: "CAPSULE",
  SYRUP: "SYRUP",
  SUSPENSION: "SUSPENSION",
  INJECTION: "INJECTION",
  CREAM: "CREAM",
  OINTMENT: "OINTMENT",
  GEL: "GEL",
  DROPS: "DROPS",
  INHALER: "INHALER",
  POWDER: "POWDER",
  SOLUTION: "SOLUTION",
  LOTION: "LOTION",
  SUPPOSITORY: "SUPPOSITORY",
  PATCH: "PATCH",
};

export const DOSAGE_FORM_LABELS = {
  [DOSAGE_FORM.TABLET]: "Tablet",
  [DOSAGE_FORM.CAPSULE]: "Capsule",
  [DOSAGE_FORM.SYRUP]: "Syrup",
  [DOSAGE_FORM.SUSPENSION]: "Suspension",
  [DOSAGE_FORM.INJECTION]: "Injection",
  [DOSAGE_FORM.CREAM]: "Cream",
  [DOSAGE_FORM.OINTMENT]: "Ointment",
  [DOSAGE_FORM.GEL]: "Gel",
  [DOSAGE_FORM.DROPS]: "Drops",
  [DOSAGE_FORM.INHALER]: "Inhaler",
  [DOSAGE_FORM.POWDER]: "Powder",
  [DOSAGE_FORM.SOLUTION]: "Solution",
  [DOSAGE_FORM.LOTION]: "Lotion",
  [DOSAGE_FORM.SUPPOSITORY]: "Suppository",
  [DOSAGE_FORM.PATCH]: "Patch",
};

export const DOSAGE_FORM_OPTIONS = Object.entries(DOSAGE_FORM_LABELS).map(
  ([value, label]) => ({
    value,
    label,
  })
);
